### DISCUSSION TITLE
AMBER atom names and common charge files

### DISCUSSION THREAD

**Discussion by Mino01** [lilyminium/psiresp#96] [Ideas]
This is the first open source RESP charge software I am aware of. Great! 
- It would be terrific with support for Amber charge files, e.g. *.in files (loadamberprep), *.off-files (loadoff) or MOL2 files (loadmol2) to be able to read charges from the Ambertools tleap module, see AMBER tleap loading command in parentheses 
- Since the SMILES input doesn't contain AMBER atom names, it would be great if this was supported for the output files e.g. PDB or MOL2 etc. So far it was possible to convert a PDB file to a MOL2 file with openbabel

**Comment by lilyminium** [comment]
Hi @Mino01, thanks for the suggestions! I'm not sure I understand the benefit to adding support for reading in charge files -- at the `loadamberprep *.in` point, wouldn't you already have charges? What would you do with psiresp?

By AMBER atom names, do you mean the standard AMBER atom names assigned according to the AMBER force field? It would be quite difficult to support conversion of SMILES to AMBER atom names, and moreover psiresp is intended to be generally force field / MD engine-agnostic. What you *could* do is harness the RDKit, MDAnalysis or OpenFF toolkit machinery to write files back out. [Writing to MOL2 can be difficult for various reasons](https://github.com/openforcefield/openff-toolkit/issues/1345); the easiest way I can think of without openbabel would be to read your PDB file with RDKit or MDAnalysis and convert to RDKit. From the RDKit molecule you can both use [psiresp to calculate charges and separately the RDKit molecule to OpenFF](https://psiresp.readthedocs.io/en/latest/examples/01_examples_resp_single_server.html#Molecules), and then use OpenFF to write it out into a mol2 file.

**Reply by Mino01** [reply]
Hi @lilyminium, 

Thanks for your reply. 
The psiresp code is very much appreciated as an open source alternative to either RESP charges derived from closed source commercial Gaussian or Turbomole QM codes. My suggestions is based on my experience with molecular dynamics simulation and protein–ligand binding in drug discovery that is my application and use of psiresp. RESP charges has become the standard charges for non-standard residues. Typically, a protein and a ligand are simulated as a complex with pristine coordinates. Non-standard residue charges is not supported by many academic commerical softwares. The charges of the protein is supported by standard force fields, whereas the force-field parameters for the non-standard residue i.e. ligand is not provided. 

RESP charges in e.g. PDB format or MOL2 format for non-standard residues such as ligands in protein–ligand complexes still require mapping to a pristine coordinate file e.g. a PDB file used together with an MD software in order to read the charges from file. The pristine coordinate file may contain the crystal structure conformation of the protein and/or the ligand or a docked structure of the protein and the ligand. 

The current atom numbers that are output from psiresp (from SMILES input) can be manually mapped to the coordinates of a pristine PDB file for use in e.g. molecular dynamics simulation (e.g. by comparing PDB file atom names and atom numbers visually, respectively for two files). However, it is a cumbersome task to manually map the atom names for 100+ atoms, and it quickly becomes a bottleneck for a large set of ligands in drug discovery. 

To keep psiresp force field / MD engine-agnostic a great feature of psiresp would be that it automates mapping of ligands for the user by mapping the charges to any input atom names provided by the user by a corresponding pristine PDB file. 

Perhaps there are some classes from Biopython e.g. Bio.PDB.Atom that can be used to extend psiresp.
